firmware/lucidgloves-firmware/haptics.ino

```diff
@@ -20,15 +33,18 @@ void setupServoHaptics(){
   thumbServo.attach(PIN_THUMB_MOTOR);
 }
 
-//static scaling, maps to entire range of servo
-void scaleLimits(int* hapticLimits, float* scaledLimits){
-  for (int i = 0; i < 5; i++){
-    scaledLimits[i] = 180.0f - hapticLimits[i] / 1000.0f * 180.0f;
-  }
+// Re-implimentation of map() using floats. If we need speed in the future swap to the integer
+// based map() function and give up a bit of accuracy.
+inline long scaleLimits(float x, float in_min, float in_max, float out_min, float out_max) {
+  // Map one range to the other.
+  x = (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
+
+  // Constrain x to the servo limits. We don't ever want to try to move to 200 degrees.
+  return (long)constrain(x, out_min, out_max);
 }
```

Comment:
does this need to return a `long`? Also I'd leave it up to the compiler to decide when to inline the function.